emcc.py

```diff
@@ -1112,7 +1112,13 @@ def add_link_flag(i, f):
       shared.Settings.EXPORT_ES6 = 1
       shared.Settings.MODULARIZE = 1
 
-    wasm_target = unsuffixed(target) + '.wasm' # might not be used
+    if shared.Settings.SIDE_MODULE or final_suffix in WASM_ENDINGS:
+      # If the user asks directly for a wasm file then this *is* the target
```

Comment:
As you mentioned in a previous incarnation of this PR `final_suffix in WASM_ENDINGS` is currently redundant here, but I like it for clarity and it means its safe against adding new WASM_ENDINGS in the future.

Reply:
We don't this far with SIDE_MODULE (see return statement just above) since there is no js generated for that case.